src/schema/schema_util.c

```diff
@@ -180,3 +180,35 @@ __wt_name_check(WT_SESSION_IMPL *session, const char *str, size_t len, bool chec
     __wt_scr_free(session, &tmp);
     return (ret);
 }
+
+/*
+ * __wt_schema_convert_file_to_table --
+ *     Convert the file name to a table metadata reference. Check if the file name has wt extension.
+ *     If so, we need to remove the wt suffix too.
+ */
+int
+__wt_schema_convert_file_to_table(WT_SESSION_IMPL *session, const char *filename, char **buf)
+{
+    WT_DECL_RET;
+    size_t bufsz;
+    const char *filep;
+
+    filep = filename;
+    WT_PREFIX_SKIP_REQUIRED(session, filep, "file:");
+    if (WT_SUFFIX_MATCH(filep, ".wt")) {
+        bufsz = strlen("table:") + strlen(filep) - strlen(".wt") + 1;
+        WT_ERR(__wt_calloc_def(session, bufsz, buf));
+        WT_ERR(
+          __wt_snprintf(*buf, bufsz, "table:%.*s", (int)(strlen(filep) - strlen(".wt")), filep));
+    } else {
+        bufsz = strlen("table:") + strlen(filep) + 1;
+        WT_ERR(__wt_calloc_def(session, bufsz, buf));
+        WT_ERR(__wt_snprintf(*buf, bufsz, "table:%s", filep));
```

Comment:
This else code is not correct for other file: suffixes. This code should use `strrchr` and not worry about what specific suffix it is because I think the code would be the same for all of them other than adjusting the length correctly. One library use of `strrchr` is in `log.c:wt_log_extract_lognum`.

Reply:
I am thinking of doing this work when on the follow-up ticket when we start to support other files for partial backup e.g. lsm, index or tiered storage objects. Currently my idea is to only support `.wt` files for now. Do you think we should change the implementation here to `strrchr` here?

Reply:
I think it would be cleaner to do the `strrchr` change sooner than later to get the structure and flow in place.